isabelle/Doc/Specification/Guarantees.thy

```diff
@@ -14,132 +14,187 @@ begin
-the assets deposited by a transaction are equal to the assets in the new state and possible external payments
-(from internal account to external party).
+\<^item> validAndPositive\_state :: @{typeof validAndPositive_state}
 \<close>
 
-text \<open>@{thm computeTransaction_preserves_assets}\<close>
+text \<open>
+The following theorem asserts that if the State is valid before computing a transaction,
+it is also valid after.
+\<close>
 
-text \<open>If calling computeTransaction yields an error, then the transaction is invalid and should not be
-part of the blockchain, preserving assets in its current state.\<close>
+text \<open>\<^bold>\<open>lemma\<close> \<^emph>\<open>computeTransaction\_preserves\_validAndPositive\_state\<close>:@{thm [display,names_short, margin=40] computeTransaction_preserves_validAndPositive_state}\<close>
+
+text \<open>The function and lemma are located in \<^emph>\<open>PositiveAccounts.thy\<close>.\<close>
 
-section \<open>Contracts Always Close\<close>
+section \<open>Finite contracts\label{sec:finite-contracts}\<close>
 
 text \<open>
+All contracts are finite and will end at a particular time. We can use the following function
+(located in Semantics.thy) to see what that time is\<close>
 
-For every Marlowe Contract there is a maximum time that allows any participant
-to create an empty transaction that will close the contract successfully.
+(* We use a hardcoded type as aliases get expanded *)
+text \<open>
+\<^item> maxTimeContract :: \isa{Contract\ {\isasymRightarrow}\ POSIXTime}\<close>
+
+text 
+\<open>
+The Blockchain doesn't have the notion of a cronjob, so it can't refund participants
+after the contract timeouts on its own. Instead we can guarantee that computing a transaction
+without inputs with a validity interval that starts after the max time will always succeed 
 \<close>
 
-text \<open>@{thm [display,names_short, margin=40] timeOutTransaction_does_not_fail}\<close>
+text \<open>\<^bold>\<open>corollary\<close> \<^emph>\<open>timeOutTransaction\_does\_not\_fail\<close>: @{thm [display,names_short, margin=60] timeOutTransaction_does_not_fail}\<close>
 
-text \<open>Moreover, that transaction empties the account and produces payments for any lingering assets\<close>
+text \<open>Moreover, we can use the function \<^emph>\<open>isClosedAndEmpty\<close> to
+assert that the previous transaction will result in a Close contract with empty accounts.\<close>
 
-text \<open>@{thm [display,names_short, margin=40] timedOutTransaction_preserves_assets}\<close>
+text \<open>
+\<^item> isClosedAndEmpty :: @{typeof isClosedAndEmpty}\<close>
 
-section \<open>Positive Accounts\<close>
+text \<open>\<^bold>\<open>theorem\<close> \<^emph>\<open>timedOutTransaction\_closes\_contract\<close>:@{thm [display,names_short, margin=60] timedOutTransaction_closes_contract}\<close>
 
-text \<open>
+text \<open>The helper and lemmas are located in \<^emph>\<open>Timeout.thy\<close>.\<close>
```

Comment:
```suggestion
text \<open>The helper function and lemmas are located in \<^emph>\<open>Timeout.thy\<close>.\<close>
```
I think it is better to say "helper function", if it is a function what you mean